projects/storefrontlib/src/cms-components/cart/validation/cart-validation-warnings-state.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { skip, take, tap } from 'rxjs/operators';
+import { RoutingService, CartModification } from '@spartacus/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CartValidationWarningsStateService {
+  constructor(protected routingService: RoutingService) {
+    this.routingService
+      .getRouterState()
+      .pipe(take(1))
+      .subscribe(
+        (routerState) => (this.navigationIdCount = routerState.navigationId)
+      );
+  }
+  NAVIGATION_SKIPS = 2;
+  navigationIdCount: number;
+
+  cartValidationResult$ = new Subject<CartModification[]>();
+  checkoutRouteActivated$ = new BehaviorSubject<boolean>(false);
```

Comment:
Is it used?

Reply:
I see it being used, but I think we can remove it and only rely on the result array.

If we have something in array we display. If not, we don't.